src/mc_control/samples/ExternalForces/src/mc_external_forces_controller.h

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2021 CNRS-UM LIRMM, CNRS-AIST JRL
+ */
+
+#pragma once
+
+#include <mc_control/api.h>
+#include <mc_control/fsm/Controller.h>
+#include <mc_control/mc_controller.h>
+
+struct MC_CONTROL_DLLAPI ExternalForcesController : public mc_control::fsm::Controller
+{
+  ExternalForcesController(mc_rbdyn::RobotModulePtr rm, double dt, const mc_rtc::Configuration & config);
+
+  void reset(const mc_control::ControllerResetData & reset_data) override;
+  bool run() override;
+
+  void supported_robots(std::vector<std::string> & out) const override
+  {
+    out = {"jvrc1"};
+  }
+
+  double t() const noexcept
```

Comment:
add `inline`

Reply:
Fixed in https://github.com/jrl-umi3218/mc_rtc/pull/95/commits/acee8bfc329557e61ad0f1ea6f031848a98cab64
Just a small question, but `inline` is missing for many methods, for examle, in `mc_tasks`: https://github.com/jrl-umi3218/mc_rtc/blob/ccb1b8567be6eccd5a3009275842693f8564ea8e/include/mc_tasks/lipm_stabilizer/StabilizerTask.h#L242-L245
Is it OK for some reason or is it just forgotten?

Reply:
Just forgotten but probably ok in practice.

I've added the missing inlines (for this task) in ee71aec81

Reply:
In the update, `noexcept` is added to most of the `inline` methods.
Some of them calls non-`noexcept` method inside (https://github.com/jrl-umi3218/mc_rtc/pull/95/commits/ee71aec81f3f5b2a93a474570f8ef1f09da99056#diff-59c59e7e81d4b42954dde739045b64f14846c73ff9dbf842e3b840609582f485R344), and some of them access to the shared pointer (https://github.com/jrl-umi3218/mc_rtc/pull/95/commits/ee71aec81f3f5b2a93a474570f8ef1f09da99056#diff-59c59e7e81d4b42954dde739045b64f14846c73ff9dbf842e3b840609582f485R459). Is it OK?

Reply:
a `noexcept` function can call a potentially throwing function (see [https://en.cppreference.com/w/cpp/language/noexcept_spec]). In fact you can even `throw` in a `noexcept` function.

Quoting the notes at the end of this page which gives a nice summary of the `noexcept` effect:
```
Note that a noexcept specification on a function is not a compile-time check; it is merely a method for a programmer to inform the compiler whether or not a function should throw exceptions. The compiler can use this information to enable certain optimizations on non-throwing functions as well as enable the noexcept operator, which can check at compile time if a particular expression is declared to throw any exceptions. For example, containers such as std::vector will move their elements if the elements' move constructor is noexcept, and copy otherwise (unless the copy constructor is not accessible, but a potentially throwing move constructor is, in which case the strong exception guarantee is waived).
```

In the cases you highlighted:
- the potentially throwing `robots(robotIndex_)` should never throw (this would mean the robot has been removed from the `Robots` instance and that's a bug
- accessing shared_ptr is noexcept ([ref](https://en.cppreference.com/w/cpp/memory/shared_ptr/operator*)) (in any case these are guaranteed not to be null in the task instance)

So we can safely declare those `noexcept` because we rely on the fact that they don't throw in the first place.